# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/nsICookie.idl
# Commit: 33f21aec0e20
# Full Hash: 33f21aec0e20d39623554816399ee76d13d7e08f
# Author: edguloien <eguloien@mozilla.com>
# Date: 2022-08-23 21:47:27
# Regressor Bug: 1713203
# File Overlap Count: 1
# Description:
#   Bug 1713203 - Added check for host, scheme and origin attributes check before cookie change broadcast to content processes r=dveditz,dragana,nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D146346
# ==============================================================================

diff -r fe25d4f35ef8 -r 33f21aec0e20 netwerk/cookie/nsICookie.idl
--- a/netwerk/cookie/nsICookie.idl	Tue Aug 23 17:30:19 2022 +0000
+++ b/netwerk/cookie/nsICookie.idl	Tue Aug 23 17:36:24 2022 +0000
@@ -10,6 +10,14 @@
  * Main cookie object interface.
  */
 
+%{C++
+namespace mozilla::net {
+class Cookie;
+}
+%}
+
+[ref] native const_Cookie(const mozilla::net::Cookie);
+
 typedef long nsCookieStatus;
 typedef long nsCookiePolicy;
 
@@ -80,6 +88,9 @@
     [implicit_jscontext]
     readonly attribute jsval originAttributes;
 
+    [noscript, notxpcom, nostdcall, binaryname(AsCookie)]
+    const_Cookie AsCookie();
+
     /**
      * true if the cookie is a session cookie.
      * note that expiry time will also be honored